Social/Facebook/Oauth.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the Integrated package.
+ *
+ * (c) e-Active B.V. <integrated@e-active.nl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Integrated\Bundle\SocialBundle\Social\Facebook;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Facebook;
+use Integrated\Bundle\SocialBundle\Oauth\OauthInterface;
+use Integrated\Common\Channel\Connector\Config\OptionsInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class Oauth
+ * @package Integrated\Bundle\SocialBundle\Social\Facebook
+ */
+class Oauth implements OauthInterface
+{
+    private $facebook_app_id;
+
+    private $facebook_app_id_secret;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * Oauth constructor.
+     * @param $facebook_app_id
+     * @param $facebook_app_id_secret
+     * @param RequestStack $requestStack
+     */
+    public function __construct($facebook_app_id, $facebook_app_id_secret, RequestStack $requestStack)
+    {
+        $this->facebook_app_id = $facebook_app_id;
+        $this->facebook_app_id_secret = $facebook_app_id_secret;
+        $this->requestStack = $requestStack->getCurrentRequest();
+    }
+
+    /**
+     * @param $connector
+     * @param $admin_url
+     * @return string
+     */
+    public function login($connector, $admin_url)
+    {
+        $fb = new Facebook([
+            'facebook_app_id' => $this->facebook_app_id,
+            'app_secret' => $this->facebook_app_id_secret,
+            'default_graph_version' => 'v2.2'
+        ]);
+
+        $helper = $fb->getRedirectLoginHelper();
+
+        $this->requestStack = $this->requestStack->createFromGlobals();
+
+        $loginUrl = $helper->getLoginUrl(
+            $this->requestStack->server->get('REQUEST_SCHEME')
+            . "://"
+            . $this->requestStack->server->get('HTTP_HOST')
+            . "/"
+            . $admin_url
+            . '/connector/config/'
+            . rawurlencode($connector),
+            ['publish_actions']
+        );
+
+        return $loginUrl;
+    }
+
+    /**
+     * @param OptionsInterface $options
+     * @return array
+     * @throws FacebookSDKException
+     * @throws \Exception
+     */
+    public function callback(OptionsInterface $options)
+    {
+        $fb = new Facebook([
+            'facebook_app_id' => $this->facebook_app_id,
+            'app_secret' => $this->facebook_app_id_secret,
+            'default_graph_version' => 'v2.2',
+        ]);
+
+        $helper = $fb->getRedirectLoginHelper();
+
+        try {
+            $accessToken = $helper->getAccessToken();
+        } catch (FacebookResponseException $e) {
```

Comment:
What is the use of turning this specific exception in a general exception with just some extra text ?